src/android/com/microsoftopentechnologies/intellij/wizards/createvm/CreateCloudServiceForm.java

```diff
@@ -0,0 +1,217 @@
+                        super.getListCellRendererComponent(jList, o, i, b, b1)
+                        : super.getListCellRendererComponent(jList, "  " + o.toString(), i, b, b1);
+            }
+        });
+
+        nameTextField.getDocument().addDocumentListener(new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent documentEvent) {
+                validateEmptyFields();
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent documentEvent) {
+                validateEmptyFields();
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent documentEvent) {
+                validateEmptyFields();
+            }
+        });
+
+        regionOrAffinityGroupComboBox.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent itemEvent) {
+                validateEmptyFields();
+            }
+        });
+
+    }
+
+    private void validateEmptyFields() {
+        boolean allFieldsCompleted = !(
+                nameTextField.getText().isEmpty() || regionOrAffinityGroupComboBox.getSelectedObjects().length == 0);
+
+        buttonOK.setEnabled(allFieldsCompleted);
+    }
+
+    public void fillFields(final Subscription subscription, Project project) {
+        this.subscription = subscription;
+
+        subscriptionComboBox.addItem(subscription.getName());
+
+        regionOrAffinityGroupComboBox.addItem("<Loading...>");
+
+        ProgressManager.getInstance().run(new Task.Backgroundable(project, "Loading regions...", false) {
+            @Override
+            public void run(ProgressIndicator progressIndicator) {
+                progressIndicator.setIndeterminate(true);
+
+                try {
+
+                    final List<AffinityGroup> affinityGroups = AzureSDKManagerImpl.getManager().getAffinityGroups(subscription.getId().toString());
+                    final List<Location> locations = AzureSDKManagerImpl.getManager().getLocations(subscription.getId().toString());
+
+                    ApplicationManager.getApplication().invokeLater(new Runnable() {
+                        @Override
+                        public void run() {
+                            final Vector<Object> vector = new Vector<Object>();
+                            vector.add("Regions");
+                            vector.addAll(locations);
+                            if (affinityGroups.size() > 0) {
+                                vector.add("Affinity Groups");
+                                vector.addAll(affinityGroups);
+                            }
+
+                            regionOrAffinityGroupComboBox.removeAllItems();
+                            regionOrAffinityGroupComboBox.setModel(new DefaultComboBoxModel(vector) {
+                                public void setSelectedItem(Object o) {
+                                    if (!(o instanceof String)) {
```

Comment:
Again, I think it'll be worthwhile to build a generalized solution for drop-downs which also have a sentinel "create" item.